lang/py/src/avro/schema.py

```diff
@@ -33,6 +33,10 @@
   A boolean; or
   Null.
 """
+from abc import ABCMeta, abstractmethod
```

Comment:
This is probably not necessary. Although it could prevent users from instantiating it, it doesn't really provide much benefit in my opinion. The abstract method can throw NotImplementedError in the base class instead.